_episodes/07-collaboration.md

```diff
@@ -0,0 +1,183 @@
+---
+title: "Collaboration and Licensing"
+teaching: 17
+exercises: 15
+questions:
+- "What are the concerns with accepting code from collaborators?"
+- "What mechanisms are there to ensure collaborators agree to license terms?"
+- "What concerns are there with using code from online forums?"
+- "Why are LLMs challenging for copyright and licensing?"
+objectives:
+- "Understand the challenges surrounding code contributed from outside the project."
+keypoints:
+- "Collaborators may be resctricted in their ability to contribute to open source projects (e.g. industrial partners) or unable to copyright their work (government employees)."
+- "You can include a Contributor License Agreement (CLA) to ensure collaborators agree to license terms prior to committing code."
+- "Stackoverflow content is licensed as CC BY-SA, which is incompatible with permissive or proprietary licenses."
+- "License and copyright around LLM-generated content is actively being litigated."
+---
+
+If you are lucky, you have an open-source project with an active user base and
+a few developers to help field issues.  As a project grows, it's possible you
+will get pull requests from people you've never met.  Even if the code looks great,
+it's possible the has author inadvertently has caused a licensing issue.
+
+A somewhat related topic is how to handle code snippets from external sources
+like web forums or large language models.
+
+## Contributors and Licenses
+
+There are two cases where a contributor can cause problems with licensing your code.
+
+First, if you have a copyleft license, contributors from industry may be reluctant
+to contribute over fear of accidentally violating the terms of the GPL license.
+Since GPL requires derivative works to be copyleft, if the contributor were to
+incorporate some of the code into a larger project, they would be obligated to
+release the entire code base under the license terms.
+
+Alternatively, if you have a restrictive license, it's possible a contribution may not be
+subject to copyright at all!  Any work from a government employee, say a researcher at a
+federal agency, is not subject to copyright and is in public domain.
+
+It is important to communicate your license terms to all collaborators and decide
+on a license early in a project's life cycle.  Changing a license is possible,
+but may require the explicit approval of all contributors; for larger, older
+projects the prospect is daunting.
+
+Part of being proactive is developing and publishing your CONTRIBUTING guidelines.
+You can also choose to use a Contributor License Agreement (CLA), which is a legal
+document that new contributors must sign prior to merging their code.  While
+protective of the software project, CLAs may limit inclusivity by acting as
+a barrier to first time contributors.  They can also create a power imbalance between
```

Comment:
And as a practical matter, CLAs often require review and approval by the legal department of a would-be contributor.